drivers/stepper/step_dir/step_dir_stepper_counter_timing.h

```diff
@@ -0,0 +1,20 @@
+/*
+ * SPDX-FileCopyrightText: Copyright (c) 2024 Fabian Blatz <fabianblatz@gmail.com>
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_DRIVER_STEPPER_STEP_DIR_STEPPER_COUNTER_TIMING_H_
+#define ZEPHYR_DRIVER_STEPPER_STEP_DIR_STEPPER_COUNTER_TIMING_H_
+
+#include "step_dir_stepper_timing_source.h"
+
+int step_counter_timing_source_init(const struct device *dev);
+int step_counter_timing_source_update(const struct device *dev, const uint32_t velocity);
+int step_counter_timing_source_start(const struct device *dev);
+bool step_counter_timing_source_needs_reschedule(const struct device *dev);
+int step_counter_timing_source_stop(const struct device *dev);
+bool step_counter_timing_source_is_running(const struct device *dev);
```

Comment:
Do these functions need to be exposed?

Reply:
True, they do not need to be exposed. Removed the headers :^)